include/seastar/rpc/rpc.hh

```diff
@@ -96,7 +97,29 @@ struct resource_limits {
     size_t max_memory = rpc_semaphore::max_counter(); ///< Maximum amount of memory that may be consumed by all requests
     /// Configures isolation for a connection based on its isolation cookie. May throw,
     /// in which case the connection will be terminated.
-    std::function<isolation_config (sstring isolation_cookie)> isolate_connection = default_isolate_connection;
+    using syncronous_isolation_function = std::function<isolation_config (sstring isolation_cookie)>;
+    using asyncronous_isolation_function = std::function<future<isolation_config> (sstring isolation_cookie)>;
+    class isolate_connection_wrapper : public  std::variant<syncronous_isolation_function, asyncronous_isolation_function> {
```

Comment:
Use composition instead of inheritance. Standard types are not meant to be inherited from usually.

Reply:
moved to composition :slightly_smiling_face: